content/docs/introduction/autoscaling.md

```diff
@@ -0,0 +1,65 @@
+
+### NeonVM
+
+Kubernetes does not support the creation or management of VMs. To address this limitation, Neon uses a tool called [NeonVM](/docs/reference/glossary#neonvm). This tool is a custom resource definition and controller for VMs, handling tasks such as adding or removing CPUs and memory. Internally, NeonVM utilizes [QEMU](/docs/reference/glossary#qemu) and [KVM](/docs/reference/glossary#kvm) (where available) to achieve near-native performance.
+
+When an autoscaler-agent needs to modify a VM's resource allocation, it simply updates the corresponding NeonVM object in Kubernetes, and the VM controller then manages the rest of the process.
+
+### Live migration
+
+In cases where a Kubernetes node becomes saturated, the NeonVM tool, which uses a hypervisor such as QEMU, manages the process of [live migrating](/docs/reference/glossary#live-migration) a VM, transferring the VM from one machine to another with minimal interruptions (typically around 100ms). Live migration transmits the internal state of the original VM to a new one while the former continues to operate, swiftly transitioning to the new VM once most of the data is copied. From within the VM, the only indication that a migration occurred might be a minor performance reduction. Importantly, the VM retains its IP address, ensuring that connections are preserved and queries remain uninterrupted.
+
+The live migration process allows for the proactive reduction of node load by migrating VMs away before reaching capacity. Although it is still possible for the node to fill up in the interim, Neon's separation of storage and compute means that VMs typically use minimal disk space, resulting in fast migrations.
+
+### Memory scaling
+
+Memory consumption on a VM can escalate rapidly in specific scenarios. Fortunately, Neon's Autoscaling system is able to detect memory usage increases without constantly requesting metrics from the VM. This is accomplished using [cgroups](/docs/reference/glossary#cgroup), which provide notification when memory usage crosses a specified threshold. Running PostgreSQL within a cgroup inside the VM allows for the receipt of such notifications. Using cgroups in this way requires running a [vm-informant](/docs/reference/glossary#vm-informant) in the VM alongside PostgreSQL to request more resources from the autoscaler-agent when PostgreSQL consumes too much memory and verify that downscaling requests from an autoscaler-agent will leave sufficient memory.
```

Comment:
```suggestion
PostgreSQL's memory consumption can escalate rapidly in specific scenarios. Fortunately, Neon's Autoscaling system is able to detect memory usage increases without constantly requesting metrics from the VM. This is accomplished by running PostgreSQL within a [cgroup](/docs/reference/glossary#cgroup), which provides notifications when memory usage crosses a specified threshold. Using cgroups in this way requires running our [vm-informant](/docs/reference/glossary#vm-informant) in the VM alongside PostgreSQL to request more resources from the autoscaler-agent when PostgreSQL consumes too much memory. The vm-informant also verifies that downscaling requests from an autoscaler-agent will leave sufficient memory leftover.
```

rewording.

Also always referring to it as "PostgreSQL" feels awkward, but I'd imagine there's already been plenty of bikeshedding there :)